src/main/java/edu/wpi/grip/ui/preview/PreviewsView.java

```diff
@@ -48,24 +55,106 @@ public PreviewsView(EventBus eventBus) {
     }
 
     @Subscribe
+    /**
+     * This function is called when a preview button is pushed/triggered
+     */
     public synchronized void onSocketPreviewChanged(SocketPreviewChangedEvent event) {
-        Platform.runLater(() -> {
-            final OutputSocket<?> socket = event.getSocket();
-
-            if (socket.isPreviewed()) {
-                // If the socket was just set as previewed, add it to the list of previewed sockets and add a new view for it.
-                if (!this.previewedSockets.contains(socket)) {
-                    this.previewedSockets.add(socket);
-                    this.previewBox.getChildren().add(SocketPreviewViewFactory.createPreviewView(this.eventBus, socket));
+        Platform.runLater(() -> {//Run this function on the main gui thread
+
+            final OutputSocket<?> socket = event.getSocket(); //The socket whose preview has changed
```

Comment:
RE 2: The easier our code is to read and follow, the easier and faster it is for new developers (who may not have had much experience) to get up to speed.